common/src/main/java/com/revenuecat/purchases/common/caching/DeviceCache.kt

```diff
@@ -320,9 +321,13 @@ open class DeviceCache(
     fun findKeysThatStartWith(
         cacheKey: String
     ): Set<String> {
-        return preferences.all
-            ?.filterKeys { it.startsWith(cacheKey) }
-            ?.keys ?: emptySet()
+        return try {
+            preferences.all
+                ?.filterKeys { it.startsWith(cacheKey) }
```

Comment:
it seems like the part that crashes is `it.startsWith(cacheKey)`, right? 
Getting here means that `preferences.all` returned a collection that has one item with a `null` key? Is that possible? 

If that's the case, do you think it's worth being more specific and checking that they key is not null before doing the `startsWith` call?

Reply:
nevermind, it was on line 325, scratch that entirely

Reply:
It is documented that it can throw NPE, but they failed to mention when and why
https://developer.android.com/reference/android/content/SharedPreferences#getAll()